### ISSUE TITLE
Issue with string[] option type

### ISSUE THREAD

**Issue by Darkle** [edgedb/edgedb-net#54]
**Describe the bug**
The client errors when using an array type and an option type together.

**Reproduction**
With the following schema:
```
module default {
  type Post {
    myOptionalArrayOfStrings: array<str>;
  }
}
```
and the following code:
```fs

[<CLIMutable>]
type Post =
    { myOptionalArrayOfStrings: string[] option }

dbClient.QuerySingleAsync<Post>("select Post{*} limit 1")
```
the client errors with the following message:
```
Error: System.NotSupportedException: Cannot find inner type of Microsoft.FSharp.Core.FSharpOption`1[System.String[]]
   at EdgeDB.TypeExtensions.GetWrappingType(Type type)
   at EdgeDB.Binary.Codecs.TypeVisitor.VisitCodec(ICodec& codec)
   at EdgeDB.Binary.Codecs.CodecVisitor.Visit(ICodec& codec)
   at EdgeDB.Binary.Codecs.TypeVisitor.VisitCodec(ICodec& codec)
   at EdgeDB.Binary.Codecs.CodecVisitor.Visit(ICodec& codec)
   at EdgeDB.ObjectBuilder.BuildResult[TType](EdgeDBBinaryClient client, ICodec codec, Data& data)
   at EdgeDB.EdgeDBBinaryClient.QuerySingleAsync[TResult](String query, IDictionary`2 args, Nullable`1 capabilities, CancellationToken token)
   at EdgeDB.EdgeDBClient.QuerySingleAsync[TResult](String query, IDictionary`2 args, Nullable`1 capabilities, CancellationToken token)
   at EdgeDB.EdgeDBClient.QuerySingleAsync[TResult](String query, IDictionary`2 args, Nullable`1 capabilities, CancellationToken token)
   at Program.Pipe #2 input at line 23@23.MoveNext() in /home/coop/Coding/edgedb-temp/Program.fs:line 25

```
Minimal repro here: https://github.com/Darkle/edgedb-temp

**Versions (please complete the following information):**

- OS: Debian Linux
- EdgeDB version: 3.0+e7d38e9
- EdgeDB CLI version: 3.3.0+071876c
- `edgedb-net` version: 1.2.1 
- .NET version: 7.0.305

**Comment by quinchs** [comment]
Seems like the cause is the option type, the binding doesn't unwrap them. Fixing this momentarily

**Comment by quinchs** [comment]
Fixed in #58
